alf/bin/train.py

```diff
@@ -67,54 +71,149 @@ def _define_flags():
     flags.DEFINE_multi_string('conf_param', None, 'Config binding parameters.')
     flags.DEFINE_bool('store_snapshot', True,
                       'Whether store an ALF snapshot before training')
+    flags.DEFINE_enum('distributed', 'none', ['none', 'multi-gpu'],
+                      'Whether store an ALF snapshot before training')
+    flags.mark_flag_as_required('root_dir')
 
 
 FLAGS = flags.FLAGS
 
 
-@alf.configurable
-def train_eval(root_dir):
-    """Train and evaluate algorithm
+def _setup_logging(rank: int = 0, log_dir: Optional[str] = None):
+    """Setup logging for each process
 
     Args:
-        root_dir (str): directory for saving summary and checkpoints
+        rank: The ID of the process among all of the DDP processes
+        log_dir: path to the direcotry where log files are written to
     """
-    trainer_conf = policy_trainer.TrainerConfig(root_dir=root_dir)
-    if trainer_conf.ml_type == 'rl':
-        trainer = policy_trainer.RLTrainer(trainer_conf)
-    elif trainer_conf.ml_type == 'sl':
-        trainer = policy_trainer.SLTrainer(trainer_conf)
-    else:
-        raise ValueError("Unsupported ml_type: %s" % trainer_conf.ml_type)
+    FLAGS.alsologtostderr = True
+    logging.set_verbosity(logging.INFO)
+
+    # When there are multiple processes, only the process with rank = 0 writes
+    # to the log directory to avoid race condition.
+    if rank == 0 and log_dir is not None:
+        logging.get_absl_handler().use_absl_log_file(log_dir=log_dir)
```

Comment:
I think process id is the suffix of the name of absl log file. So there should be no race condition.

Reply:
Verified that there are suffix of the process together with the timestamp, so we should be fine. Thanks!

I removed this piece of code.